central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -121,17 +123,21 @@ func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *stora
 		}
 	}
 
-	// Check if scan configuration already exists.
-	found, err := m.scanSettingDS.ScanConfigurationExists(ctx, scanRequest.GetScanConfigName())
-	if err != nil {
-		log.Error(err)
-		return nil, errors.Wrapf(err, "Unable to create scan configuration named %q.", scanRequest.GetScanConfigName())
-	}
-	if found {
-		return nil, errors.Errorf("Scan configuration named %q already exists.", scanRequest.GetScanConfigName())
-	}
+	// No ID means an add so we need to make sure the name does not already exist
+	if scanRequest.Id == "" {
+		// Check if scan configuration already exists by name.
+		found, err := m.scanSettingDS.ScanConfigurationExists(ctx, scanRequest.GetScanConfigName())
+		if err != nil {
+			log.Error(err)
+			return nil, errors.Wrapf(err, "Unable to create scan configuration named %q.", scanRequest.GetScanConfigName())
+		}
+		if found {
+			return nil, errors.Errorf("Scan configuration named %q already exists.", scanRequest.GetScanConfigName())
+		}
 
-	scanRequest.Id = uuid.NewV4().String()
+		scanRequest.Id = uuid.NewV4().String()
+		createScanRequest = true
+	}
 	scanRequest.CreatedTime = types.TimestampNow()
 	err = m.scanSettingDS.UpsertScanConfiguration(ctx, scanRequest)
```

Comment:
This will create a new config if I use an update request with scanRequest.Id filled. Do you have a check for that?

Reply:
good point.  need to make sure the config exists if we are doing an update.